From 35514e32ad268d80d00a5cc235bf7f857dba94d6 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Thu, 25 Jul 2024 23:36:11 +1100
Subject: [PATCH 16/16] don't mess around with ProfilerLabel being defined or
 not

src/third_party/mozjs/SConscript does set the required macro IMPL_MFBT,
but for some reason this doesn't seem to be picked up
---
 src/third_party/mozjs/include/mozilla/AutoProfilerLabel.h | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/src/third_party/mozjs/include/mozilla/AutoProfilerLabel.h b/src/third_party/mozjs/include/mozilla/AutoProfilerLabel.h
index 82299c7b157..9fa4d129c6a 100644
--- a/src/third_party/mozjs/include/mozilla/AutoProfilerLabel.h
+++ b/src/third_party/mozjs/include/mozilla/AutoProfilerLabel.h
@@ -37,10 +37,6 @@ typedef void (*ProfilerLabelExit)(void* EntryContext);
 MFBT_API void RegisterProfilerLabelEnterExit(ProfilerLabelEnter aEnter,
                                              ProfilerLabelExit aExit);
 
-// This #ifdef prevents this AutoProfilerLabel from being defined in libxul,
-// which would conflict with the one in the profiler.
-#ifdef IMPL_MFBT
-
 class MOZ_RAII AutoProfilerLabel {
  public:
   AutoProfilerLabel(const char* aLabel, const char* aDynamicString);
@@ -64,8 +60,6 @@ inline bool IsValidProfilerLabel(const ProfilerLabel& aLabel) {
   return !!std::get<0>(aLabel);
 }
 
-#endif
-
 }  // namespace mozilla
 
 #endif  // mozilla_AutoProfilerLabel_h
